action/flow/definition/mapper.go

```diff
@@ -104,25 +102,29 @@ func (m *BasicMapper) Apply(inputScope data.Scope, outputScope data.Scope) error
 		case data.MtAssign:
 
 			attrName, attrPath, pathType := data.GetAttrPath(mapping.Value)
-
-			tv, exists := inputScope.GetAttr(attrName)
-			attrValue := tv.Value
-
-			if exists && len(attrPath) > 0 {
-				if tv.Type == data.PARAMS {
-					valMap := attrValue.(map[string]string)
-					attrValue, exists = valMap[attrPath]
-				} else if tv.Type == data.ARRAY && pathType == data.PT_ARRAY {
-					//assigning part of array
-					idx, _ := strconv.Atoi(attrPath)
-					//todo handle err
-					valArray := attrValue.([]interface{})
-					attrValue = valArray[idx]
-				} else {
-					//for now assume if we have a path, attr is "object"
-					valMap := attrValue.(map[string]interface{})
-					attrValue = data.GetMapValue(valMap, attrPath)
-					//attrValue, exists = valMap[attrPath]
+            var attrValue interface{}
+            tv, exists := inputScope.GetAttr(attrName)
+			if tv == nil && pathType == data.PT_PROPERTY {
+				// Property resolution
+				attrValue, exists = property.Resolve(attrPath)
```

Comment:
So what happens if the property does not exist? like for example the Env Variable has not been set?